receiver/otlpreceiver/factory_test.go

```diff
@@ -261,6 +261,7 @@ func TestCreateMetricReceiver(t *testing.T) {
 			assert.NoError(t, err)
 			if tt.wantStartErr {
 				assert.Error(t, mr.Start(context.Background(), componenttest.NewNopHost()))
+				assert.NoError(t, mr.Shutdown(context.Background()))
```

Comment:
Do we need to close shutdown if it returned error? Why?

Reply:
The test case that's causing a leak (`invalid_http_port`) is successfully starting a GRPC server with the default config, but failing to start a http server because of an invalid port. The HTTP server causes the `mr.Start` call to fail and return an error, but the GRPC server is up and running waiting for requests.

Upon further investigation, this should be handled by the receiver itself, not a caller since the collector fails to start on this error, the `Shutdown` call should always be called on an unsuccessful `Start` call. I've removed this call from tests and moved it into the `Start` function on error. Let me know if there's a better way to handle this.